src/support.rs

```diff
@@ -0,0 +1,105 @@
+use crate::prelude::*;
+
+/// === BASIC TYPES ===
+///
+
+#[derive(Debug, Clone)]
+pub struct DerivationPath;
+
+#[derive(Debug, Clone)]
+pub struct FactorSourceID;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum FactorSourceKind {
+    Device,
+    Ledger,
+}
+impl FactorSourceKind {
+    pub fn supports_parallelism(&self) -> bool {
+        match self {
+            Self::Device => true,
+            Self::Ledger => false,
```

Comment:
`Arculus` will not support parallelism either. Right now, I dont know if any FactorSource other than `Device` will!